sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/v2/GpuParquetScan.scala

```diff
@@ -0,0 +1,230 @@
+
+import java.net.URI
+
+import scala.collection.JavaConverters._
+
+import com.nvidia.spark.rapids.{Arm, ColumnarPartitionReaderWithPartitionValues, GpuMetric, GpuParquetFileFilterHandler, GpuParquetMultiFilePartitionReaderFactory, GpuParquetPartitionReaderFactory, GpuParquetScanBase, GpuRowToColumnConverter, ParquetPartitionReader, PartitionReaderWithBytesRead, RapidsConf}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.parquet.format.converter.ParquetMetadataConverter.{NO_FILTER, SKIP_ROW_GROUPS}
+import org.apache.parquet.hadoop.metadata.{FileMetaData, ParquetMetadata}
+
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.util.RebaseDateTime.RebaseSpec
+import org.apache.spark.sql.connector.expressions.aggregate.Aggregation
+import org.apache.spark.sql.connector.read.{InputPartition, PartitionReader, PartitionReaderFactory}
+import org.apache.spark.sql.execution.datasources.{AggregatePushDownUtils, DataSourceUtils, PartitionedFile, PartitioningAwareFileIndex}
+import org.apache.spark.sql.execution.datasources.parquet.{ParquetFooterReader, ParquetOptions}
+import org.apache.spark.sql.execution.datasources.parquet.rapids.shims.v2.GpuParquetUtils
+import org.apache.spark.sql.execution.datasources.v2.{FilePartitionReaderFactory, FileScan}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.util.SerializableConfiguration
+
+
+
+
+case class GpuParquetPartitionReaderFactoryForAggrerationPushdown(
+    @transient sqlConf: SQLConf,
+    broadcastedConf: Broadcast[SerializableConfiguration],
+    dataSchema: StructType,
+    readDataSchema: StructType,
+    partitionSchema: StructType,
+    filters: Array[Filter],
+    aggregation: Option[Aggregation],
+    parquetOptions: ParquetOptions,
+    @transient rapidsConf: RapidsConf,
+  metrics: Map[String, GpuMetric]) extends FilePartitionReaderFactory with Arm with Logging {
+  private val isCaseSensitive = sqlConf.caseSensitiveAnalysis
+  private val debugDumpPrefix = rapidsConf.parquetDebugDumpPrefix
+  private val maxReadBatchSizeRows = rapidsConf.maxReadBatchSizeRows
+  private val maxReadBatchSizeBytes = rapidsConf.maxReadBatchSizeBytes
+  private val datetimeRebaseModeInRead = parquetOptions.datetimeRebaseModeInRead
+
+  private val filterHandler = GpuParquetFileFilterHandler(sqlConf)
+
+  private def getFooter(file: PartitionedFile): ParquetMetadata = {
+    val conf = broadcastedConf.value.value
+    val filePath = new Path(new URI(file.filePath))
+
+    if (aggregation.isEmpty) {
```

Comment:
What's the point of instantiating this class with empty aggregations?  Spark has these checks because it doesn't have a separate factory for aggregation pushdown, but since this is using a separate class, why have the extra checks here?  Seems like we should simply put a `require(aggregation.nonEmpty)` at the top of this class, document it in the scaladoc for this class, and never expect someone to try to use this when there are no aggregation filters.

This would simplify the class and also reduce the number of parameters needed to be passed to this class.

Reply:
You are right! I will fix it.